azure-devops/provision-image.ps1

```diff
@@ -20,155 +47,139 @@ $PythonUrl = 'https://www.python.org/ftp/python/3.8.2/python-3.8.2-amd64.exe'
-        [string]$installerPath = $randomRoot + '.exe'
-        curl.exe -L -o $installerPath $Url
-
-        Write-Output 'Installing LLVM...'
-        $proc = Start-Process -FilePath $installerPath -ArgumentList @('/S') -NoNewWindow -Wait -PassThru
-        PrintMsiExitCodeMessage $proc.ExitCode
-    }
-    catch
-    {
-        Write-Output "Failed to install LLVM!"
-        Write-Output $_.Exception.Message
-        exit -1
-    }
+Function InstallZip {
+  Param(
+    [String]$Name,
+    [String]$Url,
+    [String]$Dir
+  )
+
+  try {
+    Write-Output "Downloading $Name..."
+    [string]$randomRoot = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
+    [string]$zipPath = $randomRoot + '.zip'
+    curl.exe -L -o $zipPath $Url
+
+    Write-Output "Installing $Name..."
+    Expand-Archive -Path $zipPath -DestinationPath $Dir -Force
+  }
+  catch {
+    Write-Output "Failed to install $Name!"
+    Write-Output $_.Exception.Message
+    exit -1
+  }
 }
 
-Function InstallPython
-{
-    Param(
-        [String]$Url
-    )
+Function InstallLLVM {
+  Param(
+    [String]$Url
+  )
 
-    Write-Output 'Downloading Python...'
+  try {
+    Write-Output 'Downloading LLVM...'
     [string]$randomRoot = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
     [string]$installerPath = $randomRoot + '.exe'
     curl.exe -L -o $installerPath $Url
 
-    Write-Output 'Installing Python...'
-    $proc = Start-Process -FilePath $installerPath -ArgumentList `
-        @('/passive', 'InstallAllUsers=1', 'PrependPath=1', 'CompileAll=1') -Wait -PassThru
-    $exitCode = $proc.ExitCode
-    if ($exitCode -eq 0)
-    {
-        Write-Output 'Installation successful!'
-    }
-    else
-    {
-        Write-Output "Installation failed! Exited with $exitCode."
-        exit $exitCode
-    }
+    Write-Output 'Installing LLVM...'
+    $proc = Start-Process -FilePath $installerPath -ArgumentList @('/S') -NoNewWindow -Wait -PassThru
+    PrintMsiExitCodeMessage $proc.ExitCode
+  }
+  catch {
+    Write-Output "Failed to install LLVM!"
+    Write-Output $_.Exception.Message
+    exit -1
+  }
+}
+
+Function InstallPython {
+  Param(
+    [String]$Url
+  )
+
+  Write-Output 'Downloading Python...'
+  [string]$randomRoot = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
+  [string]$installerPath = $randomRoot + '.exe'
+  curl.exe -L -o $installerPath $Url
+
+  Write-Output 'Installing Python...'
+  $proc = Start-Process -FilePath $installerPath -ArgumentList `
+  @('/passive', 'InstallAllUsers=1', 'PrependPath=1', 'CompileAll=1') -Wait -PassThru
+  $exitCode = $proc.ExitCode
+  if ($exitCode -eq 0) {
+    Write-Output 'Installation successful!'
+  }
+  else {
+    Write-Output "Installation failed! Exited with $exitCode."
+    exit $exitCode
+  }
 }
 
+
+Write-Output "AdminUser password not supplied; assuming already running as AdminUser"

```

Comment:
Can/should this verify something like `%USERNAME%`?

Reply:
No, it must not be that because here we're running as `NT AUTHORITY\SYSTEM` (that's the whole reason we need to do this dance in the first place)

Reply:
Oh, I see, this is in the "not elevating" branch; I think saying AdminUser is good still because it helps explain the explosion that will happen if you try to run the script unelevated :)